### PR TITLE
Setup slash commands

### LINKED ISSUE #10: Use DiscordSlash commands for the bot.
It is best to catch this before any development on the bot so we can approach it as a webhook. 
Playing into a webhook allows us to run this alongside the FastAPI.

For this, I want all the current commands on the bot converted to DiscordSlash commands.
This will overwrite the current implementation.

### PR DESCRIPTION
**Pull request by Slushs**
Changed discord.py cog and __main__ to use the discord-py-slash-commands library. + added dpy slash commands lib to req.txt.
closes #10

### CODE REVIEW

File: src/bot/__main__.py [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
 from constants import environment
+from discord_slash import SlashCommand
 
 from arena_bot import ArenaBot
 
-
-bot = ArenaBot(command_prefix="!")
-
+bot = commands.Bot(command_prefix="!")
+slash = SlashCommand(bot, override_type=True)
```
Comment (GDWR) [inline src/bot/__main__.py:12]: What is `override_type`?
Why had it been set to true?

File: src/bot/requirements.txt [hunk 2]
```diff
@@ -9,3 +9,5 @@ multidict==5.1.0
 python-dotenv==0.17.0
 typing-extensions==3.7.4.3
 yarl==1.6.3
+discord-py-slash-command==1.1.2
```
Comment (GDWR) [inline src/bot/requirements.txt:12]: Is the old discord dependency still required?

**Comment by GDWR** [comment]
Good stuff, I'll merge this when you've responded to my comments.

**Review by GDWR — APPROVED** [review]

**Comment by GDWR** [comment]
I have just added linting to any new Pull requests as an action. 
As this was made before had the trigger wasn't triggered. 

Please can make sure that your changes meet our flake8 requirements by running
`pip3 install flake8`
`flake8 src/api`
`flake8 src/bot`

  || I think `flake8 .` would also work but I prefer separating them ||
There has already been some changes made to meet flake8 in `main`

File: src/bot/requirements.txt (hunk 2 above)
Comment (Slushs) [reply src/bot/requirements.txt:12]: Yes, the old dependency is still needed alongside the new one.

File: src/bot/__main__.py (hunk 1 above)
Comment (Slushs) [reply src/bot/__main__.py:12]: We are overriding the bot prefix because we need to set a bot prefix to run the bot.

**Pull request closed without merging**